src/NHibernate/Cfg/Mappings.cs

```diff
@@ -260,7 +309,7 @@ public void AddImport(string className, string rename)
 
 		public Table AddTable(string schema, string catalog, string name, string subselect, bool isAbstract, string schemaAction)
 		{
-			string key = subselect ?? dialect.Qualify(catalog, schema, name);
+			string key = subselect ?? BuildTableNameKey(catalog, schema, name);
```

Comment:
`key` here is an internal thing and is not exposed outside, so we are not concerning by dialect specific qualification of it (eg `catalog.schema.table` in SQL Server vs `catalog_schema_table` in SQLite and SQL Server Compact).